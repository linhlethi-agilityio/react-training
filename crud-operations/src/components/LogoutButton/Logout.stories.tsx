import type { Meta, StoryObj } from '@storybook/react';

// Components
import LogoutButton from '.';

const meta: Meta<typeof LogoutButton> = {
  component: LogoutButton,
};

export default meta;
type Story = StoryObj<typeof LogoutButton>;

export const Default: Story = {
  args: {
    text: 'Logout',
    isShorter: false,
    onLogout: () => null,
  },
};

export const Shorter: Story = {
  args: {
    text: 'Logout',
    isShorter: true,
    onLogout: () => null,
  },
};
