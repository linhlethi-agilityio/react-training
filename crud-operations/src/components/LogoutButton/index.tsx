import { memo } from 'react';
import { Button, Icon } from '@chakra-ui/react';

// Components
import { LogoutIcon } from '@components';

interface LogoutButtonProps {
  text?: string;
  isShorter?: boolean;
  onLogout: () => void;
}

const LogoutButton = ({ text = 'Logout', isShorter = false, onLogout }: LogoutButtonProps) => (
  <Button
    aria-label="logout"
    data-testId="logout-button"
    rightIcon={<Icon as={LogoutIcon} ml={isShorter ? 0 : 15} />}
    variant="ghost"
    onClick={onLogout}
  >
    {!isShorter && text}
  </Button>
);

export default memo(LogoutButton);
