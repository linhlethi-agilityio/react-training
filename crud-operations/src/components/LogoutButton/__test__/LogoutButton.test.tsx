import { render } from '@test-utils';
import '@testing-library/jest-dom';

// Components
import LogoutButton from '..';

const mockProps = {
  text: 'Logout',
  isShorter: false,
  onLogout: jest.fn(),
};

describe('LogoutButton component', () => {
  test('should match snapshot for LogoutButton', () => {
    const { container } = render(<LogoutButton {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
