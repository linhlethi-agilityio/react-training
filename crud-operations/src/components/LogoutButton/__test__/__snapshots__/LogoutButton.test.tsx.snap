// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton component should match snapshot for LogoutButton 1`] = `
<div>
  <button
    aria-label="logout"
    class="chakra-button css-4xx2wk"
    data-testid="logout-button"
    type="button"
  >
    Logout
    <span
      class="chakra-button__icon css-1hzyiq5"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-1jyhgcw"
        focusable="false"
        height="13"
        viewBox="0 0 17 13"
        width="17"
      >
        <path
          d="M5.313 5.208c0-.292.239-.53.53-.53h4.782V1.558c0-.235.286-.355.452-.189l4.701 4.751c.21.21.21.545 0 .754l-4.701 4.752a.265.265 0 0 1-.452-.19V8.32H5.844a.533.533 0 0 1-.532-.532v-2.58Zm-1.063 0v2.58c0 .88.714 1.594 1.594 1.594h3.718v2.055c0 1.179 1.428 1.777 2.265.94l4.705-4.748a1.595 1.595 0 0 0 0-2.258L11.827.62c-.833-.834-2.264-.243-2.264.94v2.055h-3.72c-.88 0-1.593.717-1.593 1.593ZM0 1.718v9.563c0 .88.714 1.594 1.594 1.594h4.383a.4.4 0 0 0 .398-.398v-.266a.4.4 0 0 0-.398-.399H1.594a.533.533 0 0 1-.532-.53V1.718c0-.292.24-.532.532-.532h4.383A.4.4 0 0 0 6.375.79V.523a.4.4 0 0 0-.398-.398H1.594C.714.125 0 .839 0 1.719Z"
          fill="#000000"
        />
      </svg>
    </span>
  </button>
</div>
`;
